src/System.Runtime.Analyzers/Core/NormalizeStringsToUppercase.cs

```diff
@@ -1,48 +1,101 @@
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(ToUpperRule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterCompilationStartAction(compilationStartContext =>
+            {
+                var stringType = WellKnownTypes.String(compilationStartContext.Compilation);
+                if (stringType == null)
+                {
+                    return;
+                }
+
+                var cultureInfo = compilationStartContext.Compilation.GetTypeByMetadataName("System.Globalization.CultureInfo");
+                var invariantCulture = cultureInfo?.GetMembers("InvariantCulture").OfType<IPropertySymbol>().FirstOrDefault();
+
+                // We want to flag calls to "ToLowerInvariant" and "ToLower(CultureInfo.InvariantCulture)".
+                var toLowerInvariant = stringType.GetMembers("ToLowerInvariant").OfType<IMethodSymbol>().FirstOrDefault();
+                var toLowerWithCultureInfo = cultureInfo != null ?
+                    stringType.GetMembers("ToLower").OfType<IMethodSymbol>().FirstOrDefault(m => m.Parameters.Length == 1 && m.Parameters[0].Type == cultureInfo) :
+                    null;
+
+                if (toLowerInvariant == null && toLowerWithCultureInfo == null)
+                {
+                    return;
+                }
+
+                // We want to recommend calling "ToUpperInvariant" or "ToUpper(CultureInfo.InvariantCulture)".
+                var toUpperInvariant = stringType.GetMembers("ToUpperInvariant").OfType<IMethodSymbol>().FirstOrDefault();
+                var toUpperWithCultureInfo = cultureInfo != null ?
+                    stringType.GetMembers("ToUpper").OfType<IMethodSymbol>().FirstOrDefault(m => m.Parameters.Length == 1 && m.Parameters[0].Type == cultureInfo) :
+                    null; ;
+
+                if (toUpperInvariant == null && toUpperWithCultureInfo == null)
+                {
+                    return;
+                }
+
+                compilationStartContext.RegisterOperationAction(operationAnalysisContext =>
+                {
+                    if (operationAnalysisContext.Operation.IsInvalid)
+                    {
+                        return;
+                    }
+
+                    var invocation = (IInvocationExpression)operationAnalysisContext.Operation;
+                    var method = invocation.TargetMethod;
+                    if (method.Equals(toLowerInvariant) ||
+                        (method.Equals(toLowerWithCultureInfo) &&
+                         ((IMemberReferenceExpression)invocation.ArgumentsInParameterOrder.FirstOrDefault()?.Value)?.Member == invariantCulture))
+                    {
+                        var suggestedMethod = toUpperInvariant ?? toUpperWithCultureInfo;
```

Comment:
Is it really possible that `toUpperInvariant` will be missing? If not, this will never evaluate to `toUpperWithCultureInfo`. Are there profiles where `toUpperInvariant` is missing but toUpperWithCultureInfo` is present?